package.json

```diff
@@ -50,30 +50,39 @@
     "node": ">=12.14.0"
   },
   "dependencies": {
-    "@grpc/grpc-js": "^1.1.2",
+    "@grpc/grpc-js": "^1.1.7",
     "@types/commander": "^2.12.2",
     "@types/configstore": "^4.0.0",
     "async-mutex": "^0.2.4",
+    "bluebird": "^3.7.2",
     "chalk": "^4.0.0",
     "commander": "^5.1.0",
     "configstore": "^5.0.1",
+    "connect": "^3.7.0",
     "encryptedfs": "^1.0.2",
+    "get-port": "^5.1.1",
```

Comment:
Pretty sure this is not needed. If you really do need it, you can code it yourself. But usually we either select a `0` port if we need random port. And if not, then we fail with an error if we expect the user to supply the port. If we are meant to always use a specific port. Then we can say that the user must make that port free.

What I'm thinking is that in a P2P case, does the keynode discovery also end up resolving different ports? It could be that the system only works if all keynodes are using the same port.

Reply:
I've just replaced this with a custom implementation in a few lines.